### PR TITLE
[RFC] Add postgres playbook

### PR DESCRIPTION
**Pull request by akofink**
In the interest of disconnected DB, I present: a DB box.

### CODE REVIEW

**Comment by ehelms** [comment]
Looks similar to my year old PR -- https://github.com/theforeman/forklift/pull/269

**Comment by akofink** [comment]
Very nice, indeed. Though, this is just a postgres box with no katello/candlepin configuration because I was mainly aiming at katello-devel (which doesn't yet support candlepin DB options). I think it would be really nice to have persistent data between installs and even across boxes.

It's unclear to me why your PR wasn't merged.

**Comment by ekohl** [comment]
> It's unclear to me why your PR wasn't merged.

Lack of time? We have too many open issues and PRs in this repo.

**Comment by akofink** [comment]
Well, this is an RFC, so if one of those C's is 'close it. we don't need it', then that's fine too. :sob:

File: roles/postgres/tasks/main.yml [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+      - 'postgresql'
+      - 'postgresql-server'
+      - 'python-psycopg2' # required by postgresql_* modules
+    state: 'latest'
+
+- name: 'Initialize postgresql'
+  become: true
+  command: 'postgresql-setup initdb'
+  register: command_result
+  failed_when: command_result.rc != 0 and 'Data directory is not empty!' not in command_result.stdout
```
Comment (komidore64) [inline roles/postgres/tasks/main.yml]: I think the recommended way to check for a non-zero exitstatus would be:
```
failed_when: (command_result | failed) and [...]
```

Comment (ekohl) [reply roles/postgres/tasks/main.yml]: I wonder if I like the version of @ehelms with ```creates``` better or this one.

File: roles/postgres/tasks/main.yml [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+
+- name: 'Enable postgresql on startup'
+  become: true
+  systemd:
```
Comment (evgeni) [inline roles/postgres/tasks/main.yml]: Any reason not to use the more generic `service` module here?

File: roles/postgres/tasks/main.yml [hunk 3]
```diff
@@ -0,0 +1,54 @@
...
+      regexp: 'listen_addresses = .'
+      line: "listen_addresses = '*'"
+  - name: 'Create databases'
+    postgresql_db:
+      name: "{{ item }}"
+    with_items: "{{ postgres_databases }}"
+
+- name: 'Restart postgresql'
+  become: true
+  systemd:
```
Comment (evgeni) [inline roles/postgres/tasks/main.yml]: Any reason not to use the more generic `service` module here?

File: roles/postgres/tasks/main.yml (hunk 1 above)
Comment (evgeni) [reply roles/postgres/tasks/main.yml]: I think I like `creates` too.

File: roles/postgres/tasks/main.yml [hunk 4]
```diff
@@ -0,0 +1,54 @@
...
+  block:
+  - name: 'Create the postgres user'
+    postgresql_user:
+      name: "{{ postgres_db_user }}"
+      password: "{{ postgres_db_password }}"
+  - name: 'Allow password auth to postgres'
+    when: postgres_db_password_auth
+    block:
+    - name: 'Allow passwords on subnet'
+      lineinfile:
```
Comment (evgeni) [inline roles/postgres/tasks/main.yml]: I think I'd prefer a template instead of `lineinfile` here.

**Comment by akofink** [comment]
Yeah, let's use a [pre-built container](https://hub.docker.com/r/ansible/postgresql/) instead.

**Pull request closed without merging**